### PR TITLE
fix: loss of crossOrigin value when loadMedia is called

### PR DESCRIPTION
**Pull request by amtins**
## Description

This PR fixes the deletion of the `crossOrigin` value when `loadMedia` is called.

Fixes: https://github.com/videojs/video.js/issues/8086

## Specific Changes proposed
- stores the value of `crossOrigin` before `reset` deletes it
- restores the value of `crossOrigin` after `reset` has been called
- unit test to verify that the value of crossOrigin is restored :
  - when the `crossOrigin` method is called before `loadMedia`
  - when the `crossOrigin` value is passed via the options then `loadMedia` is called
  - when the video tag contains the `crossOrigin` property then `loadMedia` is called 

## Requirements Checklist
- [X] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [X] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by amtins** [comment]
@misteroneill thank you for taking the time to review this PR 👍🏽 

> Perhaps an alternative would be calling player.crossOrigin() in the ready callback?

TL;DR Quick answer, yes it will work.

Maybe I can add a little more context regarding this PR. This PR is the result of this issue #8068 and the comments https://github.com/videojs/video.js/issues/8068#issuecomment-1378767568 and https://github.com/videojs/video.js/issues/8068#issuecomment-1379428897

According to the `loadMedia` documentation this function should `Populate the player using a MediaObject`.

However, this function does much more, it resets the player, but also deletes and rebuilds the video element. The consequence is the creation of a new playback session losing properties that are important for the developer and _interesting for the user playback experience_ (not detailed as it's out of the scope of this PR).

## The important property for the developer

`videojs` allows the `crossOrigin` parameter to be supplied as an instance `option` even though it's not [documented](https://videojs.com/guides/options/). A developer can expect this option to remain unchanged for the life of the player instance or until it is intentionally changed. Refer to 

https://github.com/videojs/video.js/blob/1491d71b26a407111104dde7dbbbc0ee49b4c0c8/src/js/player.js#L776-L777


### Demonstration

`src` and `loadMedia` produce two different behaviors depending on which one is used:


```javascript
const player videojs('player', {crossorigin : 'anonymous'});

player.el().firstChild.crossOrigin; // -> anonymous 
player.options().crossorigin; // -> anonymous 
player.crossorigin(); // -> anonymous

player.src('...')

player.el().firstChild.crossOrigin; // -> anonymous 
player.options().crossorigin; // -> anonymous 
player.crossorigin(); // -> anonymous

player.loadMedia('...')

player.el().firstChild.crossOrigin; // -> null 
player.options().crossorigin; // -> anonymous 
player.crossorigin(); // -> null

```

This means that these two functions behave very differently. `src` keeps the current playback session while `loadmedia` recreates a new playback session almost as if it were a new instance of the player, but without the `crossOrigin`, which leans more towards to a bug.

### Code references

The workflow leading to the loss of this property:

https://github.com/videojs/video.js/blob/1491d71b26a407111104dde7dbbbc0ee49b4c0c8/src/js/player.js#L1274

https://github.com/videojs/video.js/blob/866ef24b797320c397893acef37219a53aa7496d/src/js/player.js#L1125

The fix would have been more appropriate in `techOptions`, but that also involves modifying `html5.js` and in the end would not handle as many cases as the proposed fix.
https://github.com/videojs/video.js/blob/866ef24b797320c397893acef37219a53aa7496d/src/js/player.js#L1152-L1169

https://github.com/videojs/video.js/blob/866ef24b797320c397893acef37219a53aa7496d/src/js/player.js#L1196

**Comment by gkatsev** [comment]
While we're at it, should the object that loadMedia takes accept a crossOrigin property to force the value to something specific? This would be an additional feature and can come in a separate PR.

I think this change is fine as is.

**Comment by misteroneill** [comment]
I'm going to accept your comment as a second approval @gkatsev 😄
